src/tools/pub/cmd/pub-upgrade.md

```diff
@@ -153,6 +153,12 @@ Also updates `pubspec.yaml` with the new constraints.
 To check which dependencies will be upgraded,
 you can use `dart pub upgrade --major-versions --dry-run`.
 
+### `--tighten`
+
+Updates the lower bounds of dependencies in `pubspec.yaml` to match the
+current version, and returns a list of the changed constraints. 
```

Comment:
I committed it so it's ready to merge, but a couple things that can be handled later:

1. The [API docs say](https://dart.googlesource.com/pub.git/+/1ff7ed9f/lib/src/command/upgrade.dart#73) "...to match the resolved version" _singular_. Is that distinction important? My next question might explain why I think that...

2. Just for my info, is "the resolved version" clear/understandable for the average user? I don't understand the topic fully myself; we're talking about matching "the resolved version / the current version / etc" of _what_? Each of the dependencies individual most recent version (assuming most recent because that's what `upgrade` does)? 
   
    For some reason, reading the API docs [here](https://dart.googlesource.com/pub.git/+/1ff7ed9f/lib/src/command/upgrade.dart#178) ("...have their lower bound match the version in [packages]") I thought it meant that it updates all the lower bounds of the dependencies to match the version of the package that they're dependencies for... but that probably doesn't make sense.

Reply:
> The [API docs say](https://dart.googlesource.com/pub.git/+/1ff7ed9f/lib/src/command/upgrade.dart#73) "...to match the resolved version" singular. Is that distinction important? My next question might explain why I think that...

I think it depends on if you specify a specific dependency or not. Since this sentence says "dependencie**s**" and by default it upgrades all, I think "versions" makes sense in this case.

> Just for my info, is "the resolved version" clear/understandable for the average user? I don't understand the topic fully myself; we're talking about matching "the resolved version / the current version / etc" of what?

I'm not super familiar myself, but I'll leave some thoughts for now: 

This is a bit tricky, hence why you could say "the version upgraded to", but to understand what version is upgraded to, there's a resolution process to determine the latest/highest compatible version.

I think "resolution" and the version/constraint solver are topics we have to make sure we teach well, and "resolution" is likely a good candidate for the automatic linking/tooltip glossary functionality.